packages/vite/src/node/plugins/assetImportMetaUrl.ts

```diff
@@ -55,7 +56,12 @@ export function assetImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
 
           // potential dynamic template string
           if (rawUrl[0] === '`' && rawUrl.includes('${')) {
-            const ast = this.parse(rawUrl)
+            let [pureUrl, queryString = ''] = rawUrl.split('?')
+            if (queryString) {
+              pureUrl += '`'
+              queryString = '?' + queryString.replace(/`$/, '')
```

Comment:
I think we can `queryString.slice(-1)` here to avoid the regex?

Reply:
You mean `queryString.slice(0, -1)` ?